community/ndp/transport-socket/src/main/java/org/neo4j/ndp/transport/socket/ChunkedInput.java

```diff
@@ -193,7 +193,7 @@ private void ensure( int numBytes ) throws IOException
 
     private void ensureChunkAvailable() throws IOException
     {
-        if ( currentChunk == null || currentChunk.readableBytes() == 0 )
+        while ( currentChunk == null || currentChunk.readableBytes() == 0 )
```

Comment:
Just wondering on which situation would we need `while` instead of `if`

Given a list of input chunks:
```chunk0: bulabula
chunk1: 00 00
chunk2: another bulabula```